gmcs/generate.py

```diff
@@ -152,6 +154,124 @@ def get_sentences_from_lkb_output(output_file,  mrs_files, verb_preds):
 def clean_tree(tree):
     return re.sub(r'\("([^()]+)"\)', r'(@\1@)', tree).replace('"', '').replace('@', "'")
 
+def generate_sentences_ace(grammar_dir, mrs_files, verb_preds, delphin_dir, session):
+    iso = os.path.basename(grammar_dir)
+
+    # command to compile the grammar
+    compile_grammar_cmd = '/usr/local/bin/ace -G %(grammar_dir)s/%(iso)s.dat -g %(grammar_dir)s/ace/config.tdl' % {'grammar_dir':grammar_dir, 'iso':iso}
+    # print(compile_grammar_cmd)
+
+    # open a file to catch the ace output
+    output = open('ace_output'+session, 'w')
+    output.write('Compiling grammar...\n> '+compile_grammar_cmd+'\n')
+    output.flush()
+
+    ace_error = open('ace_error'+session, 'w')
+    ace_error.write('Compiling grammar...\n> '+compile_grammar_cmd+'\n')
+    ace_error.flush()
+
+    # compile the grammar
+    subprocess.run(compile_grammar_cmd.split(),
+                    stdout=output, stderr=ace_error, env=os.environ)
+
+    # rewrite the multiple mrs files to be one mrs per line
+    ace_mrs = open('ace_mrs'+session, 'w')
+    for mrs_file in mrs_files:
+        text = collapse_mrs_to_one_line(mrs_file)
+        ace_mrs.write(text+"\n")
+    ace_mrs.flush()
+    ace_mrs.close()
+
+    # command to generate sentences
+    # TODO: add -n [count] to enumerate only the top [count] results.
+    generate_sentences_cmd = '/usr/local/bin/ace -g %(grammar_dir)s/%(iso)s.dat -f -e  --disable-subsumption-test --show-realization-mrses ace_mrs%(session)s' % {'grammar_dir':grammar_dir, 'iso':iso, 'session':session, 'count':str(len(mrs_files))}
```

Comment:
It looks like `count` is not used in the command.

Reply:
It is not, I'm going to be removing it and the generate command as a whole BUT I will be bringing back the count later as a feature to specify max sentences to generate.

Reply:
Ok, you can pass `"-n"` and the count as an command argument. But note that this is how many results to generate per input. E.g., if you pass in 3 MRSs with `"-n1"`, ACE will generate up to 3 sentences. I mention this because `str(len(mrs_files))` looks like you're counting how many input MRSs there are.

Reply:
Yes thank you for the clarification.

The old code I had there was just for me to test how the count option works and I put `str(len(mrs_files))` in there in some sense as a placeholder while I was figuring it out.

In the next few commits I will be adding an option for users to specify how many sentences they want to generate. Perhaps I can set a default max though.